src/utils/waiting/waiting.go

```diff
@@ -0,0 +1,86 @@
+package waiting
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	GeneralWaitTimeoutInt int           = 30
+	GeneralWaitInterval   time.Duration = 30 * time.Second
+	WaitTimeout           time.Duration = 70 * time.Minute
+)
+
+type wait struct {
+	timeout   time.Duration
+	interval  time.Duration
+	cancelctx context.Context
+	message   string
+	predicate func() bool
+}
+
+func Wait() *wait {
+	return &wait{
+		timeout:  WaitTimeout,
+		interval: GeneralWaitInterval,
+	}
+}
+
+func (p *wait) WithPredicate(predicate func() bool) *wait {
+	p.predicate = predicate
+	return p
+}
+
+func (p *wait) WithTimeout(timeout time.Duration) *wait {
+	p.timeout = timeout
+	return p
+}
+
+func (p *wait) WithNoTimeout() *wait {
+	p.timeout = time.Duration(1<<63 - 1)
+	return p
+}
+
+func (p *wait) WithInterval(interval time.Duration) *wait {
+	p.interval = interval
+	return p
+}
+
+func (p *wait) WithCancel(cancelctx context.Context) *wait {
+	p.cancelctx = cancelctx
+	return p
+}
+
+func (p *wait) WithMessage(message string) *wait {
+	p.message = message
+	return p
+}
+
+func (p *wait) Start() (error, error, bool) {
```

Comment:
why 2 errors are needed? why does it matter if it was canceled or not? 
Cancel is coming from outside so what's point of verifying error if i just cancelled it by myself?